### PR TITLE
Add functions to get compatibility with IEEE 1788

### PR DESCRIPTION
**Pull request by lbenet**
Add atan2, and use Interval(big(...),big(...)) in ^

The second one yields the same results without the penalty of rounding.

### CODE REVIEW

**Comment by lbenet** [comment]
I've pushed some new commits with tests, which are related with what is required for conformance with the IEEE1788. I have also updated the list in #67 to refer to the concrete commit where the implementation was made. Including the last commit, I think we have all functions we needed.

@dpsanders Review carefully `cancelminus` (`cancelplus` is the same, slightly disguised), since some evaluation in extended precision is required, even for `Interval{BigFloat}`. The cases of interest are of the form `cancelminus( Interval(-1.0, a), Interval(-b, 1.0) )`, where a, b are *positive* floating points, less  than `eps(1.0)`.

**Comment by lbenet** [comment]
Mmmm... I forgot actually to include tests for `Interval{BigFloat}` for the case which requires particular attention.

**Comment by lbenet** [comment]
Pushed new tests to increase coverage, including checking two tests which were `@pending` (due to Mac issues). 

If the tests do not pass, maybe we should turn off mac checking in travis; I guess this is due to the docker version we are using (because the problem was not there before this change). Needless to say, but the tests pass locally.

**Comment by lbenet** [comment]
I turned off the support for Mac; tests pass, and coverage has been increased to ~95%

File: src/intervals/functions.jl [hunk 1]
```diff
@@ -84,7 +84,7 @@ end
         hi = hull(hi, hi1)
     else
         lo, hi = with_bigfloat_precision(53) do
-            aa = @biginterval(a)
+            aa = Interval(big(a.lo), big(a.hi)) # before: @biginterval(a)
```
Comment (dpsanders) [inline src/intervals/functions.jl]: If `a` is a `BigFloat` interval, we don't want to restrict its precision to 53 bits.

File: src/intervals/functions.jl [hunk 2]
```diff
@@ -159,26 +159,44 @@ function exp(a::Interval{Float64})
...
 end
 
-function exp(a::Interval{BigFloat})
-    isempty(a) && return a
-    @controlled_round(BigFloat, exp(a.lo), exp(a.hi))
+for fn in (:exp2, :exp10)
+    @eval begin
+        function ($fn)(a::Interval{Float64})
+            isempty(a) && return a
+            res = with_bigfloat_precision(53) do
```
Comment (dpsanders) [inline src/intervals/functions.jl]: I think we should introduce a `big53` function.
Then you can write something like the following to define this kind of function.

```julia
exp(a::Interval{Float64}) = float(exp(big53(a)))
```

File: src/intervals/trigonometric_functions.jl [hunk 3]
```diff
@@ -4,9 +4,13 @@ half_pi(::Type{Float64})  = @floatinterval(pi/2)
 two_pi(::Type{Float64})   = @floatinterval(2pi)
 half_pi(::Type{BigFloat}) = @biginterval(pi/2)
 two_pi(::Type{BigFloat})  = @biginterval(2pi)
-
 half_pi{T<:AbstractFloat}(x::T) = half_pi(T)
 
+pi_interval{T<:Real}(::Type{T}) = get_pi(Float64)
```
Comment (dpsanders) [inline src/intervals/trigonometric_functions.jl]: This definition is incorrect for `BigFloat`.

File: src/intervals/trigonometric_functions.jl [hunk 4]
```diff
@@ -4,9 +4,13 @@ half_pi(::Type{Float64})  = @floatinterval(pi/2)
...
+range_atan2{T<:Real}(::Type{T}) = Interval(-pi_interval(T).hi, pi_interval(T).hi)
+half_range_atan2(::Type{Float64}) = @floatinterval(-pi/2, pi/2)
+half_range_atan2(::Type{BigFloat}) = @biginterval(-pi/2, pi/2)
```
Comment (dpsanders) [inline src/intervals/trigonometric_functions.jl]: This is inconsistent with the above definition of `range_atan2` for `BigFloat`.

Comment (dpsanders) [inline src/intervals/trigonometric_functions.jl]: This seems to be the correct way of defining the range, in the sense that it correctly changes if the interval precision changes.

**Comment by dpsanders** [comment]
This looks good, thanks! Could you please fix the comments about the definitions of trig ranges and then I think we can merge this.

File: src/intervals/arithmetic.jl [hunk 5]
```diff
@@ -318,6 +346,38 @@ function radius(a::Interval)
     max(m-a.lo, a.hi-m)
 end
 
+# cancelplus and cancelminus
+@doc doc"""
+`cancelminus(a, b)` returns the unique interval `c` such that `b+c=a`.""" ->
+function cancelminus(a::Interval, b::Interval)
+    T = promote_type(eltype(a), eltype(b))
```
Comment (dpsanders) [inline src/intervals/arithmetic.jl]: I think you should be able to do this by just defining
```julia
function cancelminus{T}(a::Interval{T}, b::Interval{T})
```
and relying on the promotion rules we already have defined. Similarly for `fma` etc.

File: src/intervals/trigonometric_functions.jl (hunk 3 above)
Comment (dpsanders) [inline src/intervals/trigonometric_functions.jl]: We should have either `get_pi` or `pi_interval`. This function should return the current value of `interval_parameters.pi`, and all of the other `pi` functions should be defined in terms of this.

**Comment by dpsanders** [comment]
Merging so that we can carry on working on top of this.

**Pull request merged**